addons/mail/static/src/core/common/thread_model.js

```diff
@@ -195,6 +179,15 @@ export class Thread extends Record {
     modelName;
     /** @type {string} */
     module_icon;
+    open_chat_window = Record.attr(undefined, {
+        /** @this {import("models").Thread} */
+        onUpdate() {
+            if (this.open_chat_window) {
+                this.openChatWindow({ focus: true });
+                this.open_chat_window = undefined;
+            }
+        },
+    });
```

Comment:
This `open_chat_window` looks bad design: this is only used for the livechat from visitor PoV in a quite biaised way.

On website, embed or support page it's fine for it being open in chat window: that's the normal way to open a thread. In the public discuss page, however, this is an open in discuss app.

The need for this parameter feels like a modeling issue.

Reply:
There is no function change here, previous code would set `fold_state` open which had the same effect.

Reply:
But it is indeed only used in livechat, I can move the field out of mail to clarify it.